In [1]:
import numpy as np
from scipy import stats
from matplotlib import pyplot as plt

__Задача 1__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания mu с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

In [44]:
n = 256
p = 0.95
std = 16
x_mean = 80

In [45]:
p = 0.95
alpha = 1 - p

t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)

t1, t2

(-1.959963984540054, 1.959963984540054)

In [46]:
(x_mean + t1 * std / np.sqrt(n), x_mean + t2 * std / np.sqrt(n))

(78.04003601545995, 81.95996398454005)

__Ответ__: (78.0306894301508, 81.9693105698492)

__Задача 2__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. 
1. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? 
2. Найдите P-значение для данного теста.

In [28]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
alpha = 0.01
n = 10
x_mean = 200


In [29]:
std = samples.std(ddof=1)
std

4.453463071962462

In [17]:
def statistic(samples):
    return (samples.mean() - x_mean) / (std / np.sqrt(n))

In [24]:

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-3.24983554401537, 3.2498355440153697)

In [25]:
z = statistic(samples)
z

-1.0651074037450896

In [27]:
p_left = stats.t.cdf(z, df=n - 1)
p_right = 1 - stats.t.cdf(z, df=n - 1)

pvalue = 2 * min(p_left, p_right)

pvalue

0.31456660352472854

__Ответ__: 1) утверждение продавца верное, 2) Р-значение = 0.31456660352472854

__Задача 3__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

In [56]:
a = 17
alpha = 0.025
n = 100
mean = 17.5
s = 2

In [57]:
t = (mean - a) * np.sqrt(n) / s
t

2.5

In [65]:
F_kr = (1 - 2 * alpha) / 2
F_kr

0.475

По таблице Лапласа значение критической точки равно 1,96.
Т.к. наблюдаемое значение больше критической точки - гипотеза отвергается.